### PR TITLE
Fix layout issues and more

### PR DESCRIPTION
**Pull request by m4ximuel**
# Fixes
- Fix layout issues
> - Fixes an issue where 'torrent details' modal does not work properly on iOS
> - 'Vue2-perfect-scrollbar' is no longer dependent
> - A few pixels at the end of the modal's rounded corners no longer show the background color
> - When more than one modal is open, it prevents keyboard input from the dashboard

# Improvements
- You can adjust the roundness of the entire modal through the 'form' and 'formtop' elements in 'src/styles/variables.scss'.
> - 'formtop' must be defined as a larger size than 'form'. This is how to solve the problem of displaying the background color by overlapping the rendering alpha value.
- The entire 'dashboard' and the entire 'add modal' area now respond to file drag and drop.
- now can add files by dropping them in the entire area with splash screen.

### CODE REVIEW

File: src/components/Modals/ChangeLocationModal.vue
Comment (WDaan) [inline src/components/Modals/ChangeLocationModal.vue:83]: Why copy-paste this code for every single modal? That's the whole point of the Modal Mixin

File: src/components/Modals/SettingsModal/SettingsModal.vue
Comment (WDaan) [inline src/components/Modals/SettingsModal/SettingsModal.vue:8]: I'm not sure about removing the dedicated min-height... on desktop the modal height now jumps around on different tabs and that's pretty annoying. Issues were opened in the past for this very issue.
Same goes for other modals as well

File: src/components/Modals/SearchModal/SearchModal.vue
Comment (WDaan) [inline src/components/Modals/SearchModal/SearchModal.vue:102]: You forgot to remove the perfect-scrollbar in this modal

File: src/main.js
Comment (WDaan) [inline src/main.js:23]: You forgot to remove it in both SeachModal.vue & TorrentDetailModal > Peers.vue
It can also be removed as a dependency from package.json :)

File: src/components/Torrent/Torrent.vue
Comment (WDaan) [inline src/components/Torrent/Torrent.vue:4]: I really dislike the flashy blue select color

File: src/mixins/Modal.js
Comment (WDaan) [inline src/mixins/Modal.js:5]: I still think a Mixin is the best way to have common logic instead of copy-pasting the same code in every modal... thats not easily maintainable. I would prefer improving this mixin instead.

File: src/views/Dashboard.vue
Comment (WDaan) [inline src/views/Dashboard.vue:35]: This new search is pretty cool!! :)

**Review by WDaan — CHANGES_REQUESTED** [review]
Great work on the modal re-do! They look very slick.
I do have a few small remarks, please do not take these personal

Also please always restart from master when making changes instead of building on top of your current changes.
Such big PR's with lots of changes are insanely hard to review.

**Comment by m4ximuel** [comment]
delete modal not working, I'll see all

File: src/components/Modals/ChangeLocationModal.vue
Comment (m4ximuel) [reply src/components/Modals/ChangeLocationModal.vue:83]: because of closing animation.

Comment (WDaan) [reply src/components/Modals/ChangeLocationModal.vue:83]: Yes, just increase the delete timeout a little bit in the mixin. This copy/pasting for the dialog introduces a lot of extra boiler plate code

**Comment by m4ximuel** [comment]
@WDaan Please check the changes.
And please tell me the color you want for the selected torrent. Colors must be specified for future compatibility reasons.

**Comment by WDaan** [comment]
I will! As soon as I can, probably tomorrow morning.

I don't know, what was wrong with the old color? Not distinctive enough?

**Comment by m4ximuel** [comment]
The menu itself was behaving incorrectly. In addition, even though the one-click menu is not actually selected, it is expressed in the 'selected color', causing confusion to the user, so I removed the function that the menu is basically 'selected'. Therefore, the torrent list now need a 'selected color' for defined to maintain correct compatibility.

**Comment by m4ximuel** [comment]
context-menu now perfect.
Press and hold with one finger to open
It opens instantly when pressed with two fingers.
----
Two-finger touch reacts only when touching the same item, 
It is designed in consideration of various situations and respects the user's intention as the top priority.

**Comment by WDaan** [comment]
Awesome! Once I have a look at it, we can merge & release :)

**Comment by m4ximuel** [comment]
@WDaan Icons and other details do I put off to the next brunch?
There are many changes, so I think it is good to release it this time.

**Comment by WDaan** [comment]
Yes, please do make a seperate PR for those. This one is little too big allready

File: src/components/Modals/SettingsModal/SettingsModal.vue
Comment (m4ximuel) [reply src/components/Modals/SettingsModal/SettingsModal.vue:8]: It doesn't seem to be reproduced after fixing the layout.

**Comment by WDaan** [comment]
I'm looking into using maybe using [vue-js-modal](https://www.npmjs.com/package/vue-js-modal) for modals, like it's used in [this](https://github.com/ekvedaras/redis-gui) project here. This may or may not be a good way to do it, but I think it's worth trying out.

Otherwise the Modal mixin should still be refactored a little bit, so we don't have the dialog handler code for evey single dialog

**Comment by m4ximuel** [comment]
@WDaan It's really cool, but we need to review it.
Our problem is that when we issue the modal remove command on the mutation, the modal gets removed before showing the close animation.
Therefore, I'm studying possible solutions within Modal.js, respecting your opinion of disagreeing with the repeat of the current source code. I also agree it's not good way.
Returning to your opinion, the Vue.js modal is really interesting. I think this will make the 'torrent details' more useful. As for the Vue.js modal, how about testing it in a new brunch after this brunch is finished?
The current brunch will be fetched after Modal.js is improved.

**Review by WDaan — COMMENTED** [review]
Looks great! :)

What do you think about general modal height for both settings/info modal? Now the height jumps around when changing tabs, which some people find a annoying?-

**Comment by WDaan** [comment]
FYI the new right click menu is AMAZING, works like charm on both desktop & mobile!
Really great job on that one! :tada:

**Comment by m4ximuel** [comment]
@WDaan I think the modal height change is annoying because it hides sometimes the [SAVE] button in the past. However, the current version fixes that problem. and,  I think that the modal height changes proportionally to the content when the tab changes is the standard that vuetify.js pursues. If the problem is that the content height is too short and the modal size variation is large, I agree on min-height on this point. What were people dissatisfied with?

**Comment by WDaan** [comment]
Issue: #59  
It's not that a lot of people complained (VueTorrent wasn't this popular back then either) but I do agree with some of the points made. 

Maybe the size can vary indeed.... but it shoudln't jump around THAT much. I think re-introducing a minimum height, isn't a bad idea

**Comment by m4ximuel** [comment]
@WDaan I think I know what it feels like. So, how about using min-height and max-height at the same time so that the excess always uses the scroll bar? Like qbittorrent's configuration window.

**Comment by WDaan** [comment]
I think that would be the best solution :slightly_smiling_face:

**Comment by m4ximuel** [comment]
Untitled-1

I found the cause. coding...

**Comment by WDaan** [comment]
I think we're ready to merge, no?

**Comment by m4ximuel** [comment]
@WDaan All the tests for applying the new icon have been completed. Shall we apply it?

**Comment by WDaan** [comment]
Yes, but in a seperate PR!
Let's merge this one & start fresh

**Comment by LukePrior** [comment]
This looks exciting can't wait for the new icon changes to be merged and 0.9.0 to get released.

**Comment by m4ximuel** [comment]
@LukePrior , @WDaan  Soon :)

**Comment by IevgenSobko** [comment]
Impatiently waiting for a new release.
I guess it is going to skyrocket user experience.

**Comment by WDaan** [comment]
@m4ximuel I think we could do a release already, right?

**Comment by m4ximuel** [comment]
we're ready!

**Comment by IevgenSobko** [comment]
It just an awesome release.
Totally different feel on an Android device.
Rounded corners in the context menu, autofocusing for search text, working scroll in modals in a mobile browser.
And it is just the tip of the iceberg that hidden underdense this release refactoring.

**Comment by WDaan** [comment]
Thanks for the kind feedback! I'm really glad you like it. :)

Major thanks to @m4ximuel! He did a lot of great work on this one

**Pull request merged**
